HACKING.md

```diff
@@ -0,0 +1,88 @@
+# Hacking
+
+## `make-prs.py`
+
+The `make-prs.py` script (only exists in the main branch) does the following:
+
+- Get the diff from a PR;
+- find out which chapter that PR is about;
+- create branches (in your forked repo), merge the diff into all following chapters, commit, and push;
+- create PRs to the upstream.
+
+### Configuration
+
+- GITHUB_TOKEN env var is set.
+
+### Dependencies
+
+- `git` CLI
+- `tox`
+- `PyGithub`, `requests` (both are already automatically managed by `tox`)
+
+### Usage
+
+```console
+usage: make-prs.py [-h] [-u UPSTREAM_OWNER] -f FORK_OWNER [-ur UPSTREAM_REPO_NAME]
+                   [-fr FORK_REPO_NAME] -p PULL_REQUEST_NUMBER [-i]
+
+options:
+  -h, --help            show this help message and exit
+  -u UPSTREAM_OWNER, --upstream-owner UPSTREAM_OWNER
+                        The upstream owner of the repository, defaults to `canonical`.
+  -f FORK_OWNER, --fork-owner FORK_OWNER
+                        The forked repo's owner, mandatory. Normally, it should be your own
+                        GitHub user name.
+  -ur UPSTREAM_REPO_NAME, --upstream-repo-name UPSTREAM_REPO_NAME
+                        The repository name in the upstream, defaults to `juju-sdk-tutorial-k8s`.
+  -fr FORK_REPO_NAME, --fork-repo-name FORK_REPO_NAME
+                        The repository name of the forked repo, defaults to `juju-sdk-
+                        tutorial-k8s`.
+  -p PULL_REQUEST_NUMBER, --pull-request-number PULL_REQUEST_NUMBER
+                        The PR number from which you want to merge the diff into other branches.
+  -i, --ignore-conflicts
+                        Still create the PR (and following PRs) when conflicts occur in the
+                        current branch.
+```
+
+Usage examples:
+
+- upstream org: `canonical`
+- forked repo owner: `IronCore864`
+- repo name: `juju-sdk-tutorial-k8s` (both upstream and forked)
+
+Get the diff from PR No.1 in the upstream repo and merge the diff into later chapters:
+
+```bash
+$ tox -- -f IronCore864 -p 1
```

Comment:
I think this came up in the Pebble tutorial too - we should avoid having the leading `$` if possible, so that people can just use the copy button (even though here you would need to change `IronCore864` and `1`).